src/Compilers/Core/Portable/CommandLine/EditorConfig.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Represents a single EditorConfig file, see http://editorconfig.org for details about the format.
+    /// </summary>
+    internal class EditorConfig
+    {
+        // Matches EditorConfig section header such as "[*.{js,py}]", see http://editorconfig.org for details
+        private static Regex _sectionMatcher = new Regex(@"^\s*\[(([^#;]|\\#|\\;)+)\]\s*([#;].*)?$", RegexOptions.Compiled);
+        // Matches EditorConfig property such as "indent_style = space", see http://editorconfig.org for details
+        private static Regex _propertyMatcher = new Regex(@"^\s*([\w\.\-_]+)\s*[=:]\s*(.*?)\s*([#;].*)?$", RegexOptions.Compiled);
+
+        public Section GlobalSection = new Section(string.Empty);
+
+        public string Directory { get; private set; }
+
+        public IList<Section> NamedSections = new List<Section>();
+
+        private EditorConfig()
+        {
+        }
+
+        /// <summary>
+        /// Gets whether this editorconfig is a the topmost editorconfig.
+        /// </summary>
+        public bool IsRoot
+        {
+            get
+            {
+                return this.GlobalSection.Properties.ContainsKey("root") && this.GlobalSection.Properties["root"] == "true";
```

Comment:
>e"; [](start = 121, length = 3)

is `"true"` case sensitive in editro config?

Reply:
> The section names are filepath globs (case sensitive), similar to the format accepted by gitignore.

> Currently all properties and values are case-insensitive. They are lowercased when parsed.